languages/python/oso/polar/ffi.py

```diff
@@ -24,30 +24,35 @@ def load_str(self, string, filename):
         string = to_c_str(string)
         filename = to_c_str(str(filename)) if filename else ffi.NULL
         check_result(lib.polar_load(self.ptr, string, filename))
+        process_messages(self.next_message)
 
     def new_query_from_str(self, query_str):
-        return Query(
-            check_result(lib.polar_new_query(self.ptr, to_c_str(query_str), 0))
-        )
+        query = check_result(lib.polar_new_query(self.ptr, to_c_str(query_str), 0))
+        process_messages(self.next_message)
```

Comment:
I think we might have to split apart `check_result`. As currently constructed, if `check_result` raises an error we won't process any relevant messages